src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -229,7 +230,7 @@ export class AccessibleBufferWidget extends DisposableStore {
 		if (!buffer) {
 			return '';
 		}
-		const end = buffer.length;
+		const end = Math.min(Constants.MaxBufferSize, buffer.length);
 		for (let i = startLine ?? 0; i <= end; i++) {
```

Comment:
If scrollback > 1000, doesn't this mean it will not write the content in the viewport (or anything below the top 1024 lines)?

Reply:
We don't currently write the scroll-back to the buffer, but can consider doing that. This change makes it so the most recent 1024 lines are in there

Reply:
do we want to support scroll-back? if so, does that mean we'd need to save it somewhere when it goes outside of the viewport? does such a thing exist already?

Reply:
You check > `Terminal.rows`, so you do write the scrollbak. Since you're using `getLine`, you don't want to start at index 0 through 1024, but `buffer.length - 1024` through `buffer.length`?

Reply:
I run `ls -l /usr/bin`, `ls -la` a bunch, and other commands. What I see when I focus the buffer is the most recently run 1024 lines, which is what I expect. Is that also what you expect?

Reply:
The default scrollback is 1000 lines though, you're only displaying up to a maximum of line 1024. So if scrollback is 5000, it won't work when you have filled it? I haven't tested it, but I'm pretty sure this is an issue, try setting scrollback to 5000 and run `ls -lR ~` to fill it.

Reply:
won't starting at `buffer.length - 1024` in some cases be negative and in others, not include lines that we want?

Reply:
maybe i'm confused about why the max size of the buffer is 1024. is it scrollback + terminal.rows = buffer size? i'll do some logging in a bit to figure that out

Reply:
With a scrollback fill where scrollback is 5000 and rows is 20, the oldest line will be `getLine(0)` and the newest line will be `getLine(5000+20-1)`. Currently you force a maximum line index of 1024 via `Math.min(Constants.MaxBufferSize, buffer.length);`. So once the scrollback has more than ~1000 lines, you'll start losing content and only be pulling old content.